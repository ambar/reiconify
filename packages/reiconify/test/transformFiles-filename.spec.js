const fs = require('fs')
const path = require('path')
const {promisify} = require('util')
const rimraf = require('rimraf')
const {transformFiles} = require('../lib')

describe('transform-filename', () => {
  const orignalCwd = process.cwd()
  const cwd = path.resolve(__dirname, 'fixtures/transform-filename')
  const srcDir = path.resolve(cwd, 'src')
  const cleanup = async () => {
    await promisify(rimraf)(srcDir)
  }

  beforeAll(() => {
    process.chdir(cwd)
  })

  afterEach(async () => {
    await cleanup()
  })

  afterAll(() => {
    process.chdir(orignalCwd)
  })

  it('transforms icons', async () => {
    await transformFiles({
      inputs: 'icons/*.svg',
      src: true,
      srcDir,
    })

    const files = await promisify(fs.readdir)(srcDir)
    expect(files).toMatchSnapshot()
  })
})
