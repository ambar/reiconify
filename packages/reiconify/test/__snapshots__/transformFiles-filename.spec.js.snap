// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform-filename transforms icons 1`] = `
[
  "Icon.jsx",
  "MdCheck.jsx",
  "MdThumbUp.jsx",
  "index.jsx",
]
`;
